pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java

```diff
@@ -161,29 +165,45 @@ public static GroupBuildInfo run(
         // TODO: should we bother with this case?
 
         RepositoryData repo = null;
-
-        if (repoZipPath != null) {
-            repo = parseRepository(new File(repoZipPath));
+        
+        if (repoZipPath != null || context.getPigConfiguration()
+                .getFlow()
+                .getRepositoryGeneration()
+                .getStrategy() != RepoGenerationStrategy.IGNORE) {
+            if (repoZipPath != null) {
+                repo = parseRepository(new File(repoZipPath));
+            } else {
+                repo = generateRepo(removeGeneratedM2Dups, configurationDirectory, strictLicenseCheck);
+            }
+            context.setRepositoryData(repo);
+            context.storeContext();
         } else {
-            repo = generateRepo(removeGeneratedM2Dups, configurationDirectory, strictLicenseCheck);
+            log.info("Skipping Repo Generation");
         }
-        context.setRepositoryData(repo);
-        context.storeContext();
 
-        if (!skipSources) {
+        if (!(skipSources || context().getPigConfiguration()
+                .getFlow()
+                .getSourcesGeneration()
+                .getStrategy() == SourcesGenerationStrategy.IGNORE)) {
             generateSources();
 
         } else {
             log.info("Skipping Source Generation");
         }
 
-        if (!skipJavadoc) {
+        if (!(skipJavadoc || context.getPigConfiguration()
+                .getFlow()
+                .getJavadocGeneration()
+                .getStrategy() == JavadocGenerationStrategy.IGNORE)) {
             generateJavadoc();
         } else {
             log.info("Skipping Javadoc Generation");
         }
 
-        if (!skipLicenses && repo != null) {
```

Comment:
I changed it as other ones, if user doesn't specify skipLicense and strategy is not IGNORE, it's user error.